src/fetchers/tarball-fetcher.js

```diff
@@ -130,48 +131,66 @@ export default class TarballFetcher extends BaseFetcher {
     });
   }
 
-  fetchFromExternal(): Promise<FetchedOverride> {
+  async fetchFromExternal(): Promise<FetchedOverride> {
     const registry = this.config.registries[this.registry];
 
-    return registry.request(
-      this.reference,
-      {
-        headers: {
-          'Accept-Encoding': 'gzip',
-          Accept: 'application/octet-stream',
-        },
-        buffer: true,
-        process: (req, resolve, reject) => {
-          const {reporter} = this.config;
-          // should we save this to the offline cache?
-          const tarballMirrorPath = this.getTarballMirrorPath();
-          const tarballCachePath = this.getTarballCachePath();
-
-          const {validateStream, extractorStream} = this.createExtractor(resolve, reject);
-
-          const handleRequestError = res => {
-            if (res.statusCode >= 400) {
-              const statusDescription = http.STATUS_CODES[res.statusCode];
-              reject(new Error(reporter.lang('requestFailed', `${res.statusCode} ${statusDescription}`)));
-            }
-          };
-
-          req.on('response', handleRequestError);
-          req.pipe(validateStream);
-
-          if (tarballMirrorPath) {
-            validateStream.pipe(fs.createWriteStream(tarballMirrorPath)).on('error', reject);
-          }
-
-          if (tarballCachePath) {
-            validateStream.pipe(fs.createWriteStream(tarballCachePath)).on('error', reject);
-          }
-
-          validateStream.pipe(extractorStream).on('error', reject);
-        },
-      },
-      this.packageName,
-    );
+    let retriesRemaining = 2;
```

Comment:
@bestander shall we pull this out to a constant like `MAX_SERVER_ERROR_RETRIES` or something?